File: relax-scala-spring-boot-starter/src/main/scala/com/infilos/spring/config/ContextConfigure.scala
package com.infilos.spring.config

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.ApplicationContext
import org.springframework.stereotype.Component

import javax.annotation.PostConstruct

@Component
class ContextConfigure {

  @Autowired
  private val context: ApplicationContext = null

  @PostConstruct
  def construct(): Unit = {
    ContextConfigure._Configure = this
  }
}

object ContextConfigure {
  private var _Configure: ContextConfigure = _

  def inject[T](beanClass: Class[T]): T = context.getBean(beanClass)

  def context: ApplicationContext = {
    if (isSpringRunning) {
      _Configure.context
    } else {
      throw new UnsupportedOperationException("Spring context is unavailable!")
    }
  }

  def isSpringRunning: Boolean = _Configure != null

  def inject[T](beanName: String): T = context.getBean(beanName).asInstanceOf[T]
}

File: relax-java-spring-boot-starter/src/main/java/com/infilos/spring/config/ContextConfigure.java
package com.infilos.spring.config;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.ApplicationContext;
import org.springframework.stereotype.Component;

import javax.annotation.PostConstruct;
import java.util.Objects;

@Component
public class ContextConfigure {

    private static ApplicationContext staticContext;

    @Autowired
    private ApplicationContext thisContext;

    public static boolean isSpringRunning() {
        return Objects.nonNull(staticContext);
    }

    public static ApplicationContext context() {
        if (!isSpringRunning()) {
            throw new UnsupportedOperationException("Spring context is unavailable!");
        }

        return staticContext;
    }

    @SuppressWarnings("unchecked")
    public static <T> T inject(String beanName) {
        return (T) staticContext.getBean(beanName);
    }

    public static <T> T inject(Class<T> beanClass) {
        return staticContext.getBean(beanClass);
    }

    @PostConstruct
    public void construct() {
        staticContext = thisContext;
    }
}
